scripts/ci/select_runner.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+# Check if the test requires a prover
+requires_prover() {
+  local test_name=$1
```

Comment:
this ideally would be encapsulated in the same shell script that runs the things, but nbd

Reply:
Wasn't sure on how to do this. The result of `select_runner` is used to start the machine on which `e2e_test.sh` is then run on.